internal/tfengine/tfengine_test.go

```diff
@@ -0,0 +1,35 @@
+package tfengine
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+const basePath = "../../examples/tfengine"
+
+// TestExamples is a basic test for the engine to ensure it runs without error on the examples.
+func TestExamples(t *testing.T) {
+	exs := []string{"simple", "full"}
```

Comment:
Why not list the .yaml files in that directory? Then adding more examples adds auto testing.

Reply:
Done. I initially wanted to reduce the chances of something going wrong and giving a misleading pass but I added a check for at least one example being found so we are testing something.